docs/frc/1-Basics/9-Drivetrain.md

```diff
@@ -1,3 +1,300 @@
+        return false;
+    }
+```
+In `end()`, we don't need to specify anything since we don't require any actions to happen when the command ends. However, we return false in `isFinished()` to prevent the command from ending prematurely--instead, it will only end when the command is cancelled/when there is no controller input. 
+
+## RobotContainer
+
+The final part of this process is to link the command to the subsystem and bind it to a controller. 
+
+```java
+package frc.robot;
+
+import edu.wpi.first.wpilibj.Notifier;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.commands.drivetrain.ManualDriveCommand;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.subsystems.SnailSubsystem;
+import frc.robot.util.SnailController;
+
+import java.util.ArrayList;
+
+import static frc.robot.Constants.ElectricalLayout.CONTROLLER_DRIVER_ID;
+import static frc.robot.Constants.ElectricalLayout.CONTROLLER_OPERATOR_ID;
+import static frc.robot.Constants.UPDATE_PERIOD;;
+
+public class RobotContainer {
+
+    private SnailController driveController;
+    private SnailController operatorController;
+    
+    private ArrayList<SnailSubsystem> subsystems;
+    private final Drivetrain drivetrain;
+
+    ...
+```
+Since most of the needed declarations for `RobotContainer` are pre-generated by our robot template, the only line we need to add is `private final Drivetrain drivetrain` to instantiate the subsystem. 
+
+### Constructor
+
+The next step is set up our subsystem and add a default command: 
+
+```java
+    private void configureSubsystems() {
+        drivetrain = new Drivetrain();
+        drivetrain.setDefaultCommand(new ManualDriveCommand(drivetrain, driveController::getDriveForward, driveController::getDriveTurn));
+
+        subsystems = new ArrayList<>();
+        subsystems.add(drivetrain);
+    }
+```
+
+We use `setDefaultCommand()` to pass in the `ManualDriveCommand` we wrote above. Note how in the manual command, after we pass in the `Drivetrain` object, we write `driveController::getDriveForward` and `driveController::getDriveTurn` to be passed in for the `DoubleSupplier`s. Using Java's `::` operator allows to pass in the function itself and not a result of evaluating that function. 
+
+Once again, when `execute()` is run in `ManualDriveCommand`, the command will receive *constantly-updated* controller input values for the drivetrain instead of the result of evaluating the controller's functions once. 
+
+Also, `configureButtonBindings()` is left empty here since we only have one source of functionality, which falls under the default command. If we added more button bindings, then we'd have to add to this function.
+
+## Conclusion
+
+With that, we're finished! Congratulations on finishing the `Basics` section and learning how to program multiple subsystems. In the next section, you will learn about Shuffleboard and how we can use it to test our subsystems. 
```

Comment:
I think you should say we're learning sensors and shuffleboard instead of just shuffleboard